staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -182,6 +188,8 @@ func (s *store) Delete(ctx context.Context, key string, out runtime.Object, prec
 		panic("unable to convert output object to pointer")
 	}
 	key = path.Join(s.pathPrefix, key)
+	startTime := time.Now()
+	defer etcdmetrics.RecordRequestLatency("delete", typeName(out), startTime)
```

Comment:
am not a fan of all these calls. We are breaking layering here. The storage does not know about resources or kinds, yet we try to derive something like a kind from the runtime.Object.